logging/google/cloud/logging/entries.py

```diff
@@ -144,9 +145,45 @@ class ProtobufEntry(_BaseEntry):
 
     See:
     https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
+
+    :type payload: text or dict
+    :param payload: The payload passed as ``textPayload``, ``jsonPayload``,
+                    or ``protoPayload``. This also may be passed as a raw
+                    :class:`.any_pb2.Any` if the ``protoPayload`` could
+                    not be deserialized.
+
+    :type logger: :class:`google.cloud.logging.logger.Logger`
+    :param logger: the logger used to write the entry.
+
+    :type insert_id: text
+    :param insert_id: (optional) the ID used to identify an entry uniquely.
+
+    :type timestamp: :class:`datetime.datetime`
+    :param timestamp: (optional) timestamp for the entry
+
+    :type labels: dict
+    :param labels: (optional) mapping of labels for the entry
+
+    :type severity: str
+    :param severity: (optional) severity of event being logged.
+
+    :type http_request: dict
+    :param http_request: (optional) info about HTTP request associated with
+                         the entry
     """
     _PAYLOAD_KEY = 'protoPayload'
 
+    def __init__(self, payload, logger, insert_id=None, timestamp=None,
+                 labels=None, severity=None, http_request=None):
+        super(ProtobufEntry, self).__init__(
+            payload, logger, insert_id=insert_id, timestamp=timestamp,
+            labels=labels, severity=severity, http_request=http_request)
+        if isinstance(self.payload, any_pb2.Any):
+            self.payload_pb = self.payload
+            self.payload = None
+        else:
+            self.payload_pb = None
```

Comment:
So I moved this into `ProtobufEntry` alone but now all of the `ProtobufEntry`-s pay the cost of this branch (@jonparrott and @tseaver always scold me when I worry about branches)